src/test/scala/org/constellation/domain/redownload/RedownloadServiceTest.scala

```diff
@@ -1,72 +1,231 @@
 package org.constellation.domain.redownload
 
+import java.net.InetSocketAddress
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.testkit.ScalatestRouteTest
 import cats.effect.{ContextShift, IO}
 import cats.implicits._
-import org.constellation.ConstellationExecutionContext
-import org.scalatest.{FreeSpec, Matchers}
-
-class RedownloadServiceTest extends FreeSpec with Matchers {
+import constellation._
+import de.heikoseeberger.akkahttpjson4s.Json4sSupport
+import org.constellation.TestHelpers.prepareFacilitators
+import org.constellation.consensus.EdgeProcessor
+import org.constellation.consensus.EdgeProcessor.chunkSerialize
+import org.constellation.domain.snapshotInfo.SnapshotInfoChunk
+import org.constellation.p2p.{PeerAPI, PeerData}
+import org.constellation.primitives.IPManager
+import org.constellation.schema.Id
+import org.constellation.storage.RecentSnapshot
+import org.constellation.{ConstellationExecutionContext, TestHelpers}
+import org.json4s.native.Serialization
+import org.mockito.cats.IdiomaticMockitoCats
+import org.mockito.{ArgumentMatchersSugar, IdiomaticMockito}
+import org.scalatest.{BeforeAndAfter, FreeSpec, Matchers}
+import org.constellation.Fixtures.{toRecentSnapshot, toRecentSnapshotWithPrefix}
+
+class RedownloadServiceTest
+    extends FreeSpec
+    with Matchers
+    with BeforeAndAfter
+    with ScalatestRouteTest
+    with ArgumentMatchersSugar
+    with Json4sSupport
+    with IdiomaticMockito
+    with IdiomaticMockitoCats {
 
   implicit val cs: ContextShift[IO] = IO.contextShift(ConstellationExecutionContext.unbounded)
+  implicit val serialization = Serialization
+  implicit val s: ActorSystem = system
+  implicit val dao = TestHelpers.prepareMockedDAO()
+
+  val numFacilitators = 10 //needs to be even for some tests below
+  val facilitators = prepareFacilitators(numFacilitators)
+  val socketAddress = new InetSocketAddress("localhost", 9001)
+  val ipManager: IPManager[IO] = IPManager[IO]()
```

Comment:
I think that all services should be instantiated before each test, otherwise the state could be shared across the unit tests